controllers/nodefeaturediscovery_finalizers.go

```diff
@@ -0,0 +1,411 @@
+	if err != nil {
+		return err
+	}
+
+	// Attempt to delete the Role
+	err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteRole(ctx, instance.ObjectMeta.Namespace, nfdWorkerApp)
+		if err != nil {
+			return false, interpretError(err, "nfd-worker Role")
+		}
+		klog.Info("nfd-worker Role resource has been deleted.")
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Attempt to delete the ClusterRole
+	err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteClusterRole(ctx, instance.ObjectMeta.Namespace, nfdMasterApp)
+		if err != nil {
+			return false, interpretError(err, "nfd-master ClusterRole")
+		}
+		klog.Info("nfd-master ClusterRole resource has been deleted.")
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Attempt to delete the RoleBinding
+	err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteRoleBinding(ctx, instance.ObjectMeta.Namespace, nfdWorkerApp)
+		if err != nil {
+			return false, interpretError(err, "nfd-worker RoleBinding")
+		}
+		klog.Info("nfd-worker RoleBinding resource has been deleted.")
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Attempt to delete the ClusterRoleBinding
+	err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteClusterRoleBinding(ctx, instance.ObjectMeta.Namespace, nfdMasterApp)
+		if err != nil {
+			return false, interpretError(err, "ClusterRoleBinding")
+		}
+		klog.Info("ClusterRoleBinding", nfdMasterApp, "resource has been deleted.")
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Attempt to delete the Worker ServiceAccount
+	err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteServiceAccount(ctx, instance.ObjectMeta.Namespace, nfdWorkerApp)
+		if err != nil {
+			return false, interpretError(err, "worker ServiceAccount")
+		}
+		klog.Info("Worker ServiceAccount resource has been deleted.")
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Attempt to delete the Master ServiceAccount
+	err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteServiceAccount(ctx, instance.ObjectMeta.Namespace, nfdMasterApp)
+		if err != nil {
+			return false, interpretError(err, "master ServiceAccount")
+		}
+		klog.Info("Master ServiceAccount resource has been deleted.")
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// Attempt to delete the Worker config map
+	err = wait.Poll(RetryInterval, Timeout, func() (done bool, err error) {
+		err = r.deleteConfigMap(ctx, instance.ObjectMeta.Namespace, nfdWorkerApp)
+		if err != nil {
+			return false, interpretError(err, "nfd-worker config map")
+		}
+		klog.Info("nfd-worker config map resource has been deleted.")
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	// If NFD-Topology-Updater was requested
+	if instance.Spec.TopologyUpdater {
```

Comment:
Hmm, maybe it would be good to do delete this before nfd-master?

Reply:
umm makes sense, if no will start complaining about not being able to reach the server